### ISSUE TITLE
Invalid token auth relm error when running "docker login" 

### ISSUE THREAD

**Issue by Nick-Harvey** [SUSE/caasp-services#24]
Good Morning, 

When launching the chart with ingress ssl enabled (using the Silly account), and then trying to log into the docker registry via the command line, you get the following: 

> Error response from daemon: Get https://foo.bar.com/v2/: invalid token auth challenge realm: parse http://reg-portus-nginx:%!s(<nil>)/v2/token: invalid URL escape "%!s"

New to helm in general, but my suspicion is that it's a formatting issue in the glob in the registry-configmap: 

```
data:
  {{- if .Values.portus.tls.enabled }}
  REGISTRY_AUTH_TOKEN_REALM: {{ printf "https://%s:%s/v2/token" (.Values.nginx.host | default (printf "%s" (include "nginx.fullname" .))) .Values.nginx.port | quote }}
  REGISTRY_AUTH_TOKEN_SERVICE: {{ printf "%s:%s" (include "registry.fullname" .) .Values.registry.service.port | quote }}
  REGISTRY_AUTH_TOKEN_ISSUER: {{ template "portus.fullname" . }}
  REGISTRY_HTTP_TLS_KEY: "/certificates/portus.key"
  REGISTRY_HTTP_TLS_CERTIFICATE: "/certificates/portus.crt"
  REGISTRY_AUTH_TOKEN_ROOTCERTBUNDLE: "/certificates/portus.crt"
  {{- else }}
  REGISTRY_AUTH_SILLY_REALM: {{ printf "http://%s:%s/v2/token" (.Values.nginx.host | default (printf "%s" (include "nginx.fullname" .))) (.Values.nginx.service.nodePort | default .Values.nginx.port) | quote }} 
  REGISTRY_AUTH_SILLY_SERVICE: {{ printf "%s:%s" (include "registry.fullname" .) .Values.registry.service.port | quote }}
  {{- end }}

  REGISTRY_STORAGE_PROVIDER: gcs
  REGISTRY_STORAGE_BUCKET: kube-devregistry
  REGISTRY_HTTP_SECRET: Daedalus

  REGISTRY_NOTIFICATIONS_ENDPOINTS: >
```
--- Update 1
Looks like the env variable is getting set incorrectly

```
# env | grep -i token
REGISTRY_AUTH_SILLY_REALM=http://reg-portus-nginx:%!s(<nil>)/v2/token
```

**Comment by Nick-Harvey** [comment]
Setting it manually in the configmap doesn't seem to work either. 

```
REGISTRY_AUTH_SILLY_REALM: "http://reg-portus-nginx/v2/token" 
```

Results in: 

```
http://reg-portus-nginx/v2/token?account=Nick&client_id=docker&offline_token=true&service=reg-portus-registry%3A5000: dial tcp: lookup reg-portus-nginx on 8.8.8.8:53: no such host
```

yet if you ssh into the pod and do a drill on reg-portus-nginx it works and you can curl a token 

```
/ # drill reg-portus-nginx
;; ->>HEADER<<- opcode: QUERY, rcode: NXDOMAIN, id: 8661
;; flags: qr rd ra ; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 0 
;; QUESTION SECTION:
;; reg-portus-nginx.	IN	A

;; ANSWER SECTION:

;; AUTHORITY SECTION:
.	84451	IN	SOA	a.root-servers.net. nstld.verisign-grs.com. 2018010900 1800 900 604800 86400

;; ADDITIONAL SECTION:

;; Query time: 11 msec
;; SERVER: 10.55.240.10
;; WHEN: Tue Jan  9 16:48:33 2018
;; MSG SIZE  rcvd: 109
/ # ifconfig
eth0      Link encap:Ethernet  HWaddr 0A:58:0A:34:01:1B  
          inet addr:10.52.1.27  Bcast:0.0.0.0  Mask:255.255.255.0
          inet6 addr: fe80::302a:dcff:fef2:d36d%32733/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1460  Metric:1
          RX packets:604 errors:0 dropped:0 overruns:0 frame:0
          TX packets:579 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:1518741 (1.4 MiB)  TX bytes:41293 (40.3 KiB)

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1%32733/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

/ # curl reg-portus-nginx/v2/token
{"token":"eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IjVVVVI6QkFaMjpXMkRQOko3VUk6WFMyVzpYNDUzOjRXUkE6SE9KMzpSSk9TOjYyQVA6TldWTzpaRTJLIn0.eyJpc3MiOiJwb3J0dXMiLCJzdWIiOm51bGwsImF1ZCI6bnVsbCwiaWF0IjoxNTE1NTE2NTQzLCJuYmYiOjE1MTU1MTY1MzgsImV4cCI6MTUxNTUxNjg0MywianRpIjoidHNTSjZHd3dNcm9EYVI1Y2pWamVXYlFTZGZMMWNic0VuZTJvb3ZQd1gxIiwiYWNjZXNzIjpbXX0.Gyfmw0gCEtFLZHjzaq1blCURHwPaByOU__NJcc-GuFSLkjRPaOhYnOKh7nSevcYNEhpxCUozvWkY3yyk1YcNTg1_-m2JcG3uD5NLymJZebZntr3tlD7UZGU972YSfj-Z5ZJfzrmBlRU_gZgCvy4H0hOzLnRprE1BIKzG9tFE6cInoyX6EV40YjuspXiJbGg547FToQrxNC-Gv0Upjwn7wV1ANSA8FeN-oF424EScVmlvD-APl9ckcOreGCwC1QxbesMxUbtZykFzDMidrs5Q8as03tI1wy_3HHMeNgqO8_Aa5RRrlRjXMJDRElK9b08cDXER18UMRx7eQk_L8Nea0AKgT9AYFljxt2zS_a3TV9MG-MdUN8LrNT9b4g_87s1rHFX7y7nnnejH9MOV0U-SDL_6LBZcfgRxyAb9DfGT1X8C-TqqE2Jrsu2tNuXkuVxjU3-ELWPH7ytXoG6XgcAp2l0OgPdO7tyjMkO3jYv_eNytW_-skBeYENQBz2KsWomc","expires_in":300,"issued_at":"2018-01-09T16:49:03+00:00"}/
```

**Comment by bonhamjp** [comment]
Hey @Nick-Harvey thanks for pointing out the formatting issue, that does look like it is not getting set correctly. I'll look into that.

As for connecting to the registry, is "reg-portus-nginx" the name of your internal service? `REGISTRY_AUTH_SILLY_REALM` needs to be resolvable both internally, and externally, so when using ingress that address should use the same domain as the ingress endpoint. Have you set `nginx.host`? https://github.com/kubic-project/caasp-services/blob/master/contrib/helm-charts/portus/values.yaml#L139

If you already have it running you can test this out by changing the registry cm to have:

`REGISTRY_AUTH_SILLY_REALM: "http://<ingress domain>/v2/token" `

**Comment by Nick-Harvey** [comment]
I did have the ingress host configured and setting it up manually worked just fine

**Comment by stefsuse** [comment]
This issue was addressed with PR #38
